az-digital/az_quickstart
# Closes #2547 Add ability to customize publication types.

## Description
Adds a new configurable entity type called 'publication type'.

Adds a core patch from a merge request on issue [#3347343](https://drupal.org/i/3347343)

Creates a mechanism where site owners can create arbitrary publication types and map them to  [supported types in citation style language
]( https://docs.citationstyles.org/en/stable/specification.html#appendix-iii-types)

This way the list of types is editable for both the field on the publication content type and within views with exposed forms for filters.

Adds a new configuration entity for each supported type. See the [article type](modules/custom/az_publication/config/install/az_publication.type.az_article.yml) as an example


## Related issues
Closes #2547 

## How to test
Enable az_publication
Add a new publication
Select Article in Book as the type.
See what happens.


## Types of changes


### Arizona Quickstart (install profile, custom modules, custom theme)
- **Patch release changes**
- **Minor release changes**
- **Other or unknown**

### Drupal core
- **Patch release changes**
- **Minor release changes**
- **Other or unknown**

### Drupal contrib projects
- **Patch release changes**
- **Minor release changes**
- **Other or unknown**

## Checklist




Allows changing the filter from something like.
<img width="498" alt="Screenshot 2024-10-22 at 2 38 08 PM" src="https://github.com/user-attachments/assets/cf31ddac-1736-46c1-9ee3-de575d7173e8">
To a more manage-able
<img width="553" alt="image" src="https://github.com/user-attachments/assets/cf1fec89-a325-4448-b192-a67b84cd97ac">
